### PR TITLE
RAML 1.0 support

### PR DESCRIPTION
**Pull request by JanMesaric**
- [x] add Osprey with RAML 1.0 support and a basic example
- [x] support yaml-formated examples
- [x] support `examples` property
- [x] fix example
- [x] tests:
  - [x] split tests RAML 0.8 vs. RAML 1.0
  - [x] increase test coverage, e.g. `headers`, 
- [x] upgrade devDependencies

### CODE REVIEW

File: examples/api.raml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#%RAML 1.0
+title: Example API
+baseUri: http://example.com/api
+
+/test:
+  get:
+      responses:
```
Comment (jstoiko) [inline examples/api.raml]: extra indent

File: test/osprey-mock-service.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  describe('routes', function () {
+    it('should expose a function', function () {
+      expect(mockService).to.be.a('function')
+    })
+
+    it('should respond with example parameter', function () {
+      return popsicle.default('/api/test')
+        .use(server(http))
+        .then(function (res) {
+          expect(JSON.parse(res.body)).to.have.any.keys('success')
```
Comment (jstoiko) [inline test/osprey-mock-service.js]: maybe use `to.deep.equal` here to be more strict

File: test/osprey-mock-service.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          expect(res.status).to.equal(404)
+        })
+    })
+
+    it('should have empty body when no example parameter available', function () {
+      return popsicle.default('/api/noexample')
+        .use(server(http))
+        .then(function (res) {
+          expect(res.status).to.equal(200)
+          expect(res.body).to.equal('')
```
Comment (jstoiko) [inline test/osprey-mock-service.js]: `.to.be.empty`

**Review by jstoiko — CHANGES_REQUESTED** [review]

File: test/osprey-mock-service.js (hunk 2 above)
Comment (jstoiko) [reply test/osprey-mock-service.js]: why not `expect(res.body)` shouldn't `res.body` be parsed at this point?

**Review by jstoiko — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -46,9 +46,11 @@
     "finalhandler": "^0.4.0",
     "morgan": "^1.7.0",
     "negotiator": "^0.6.0",
-    "osprey": "^0.3.0",
+    "osprey": "mulesoft/osprey#raml1.0",
     "osprey-resources": "^0.7.0",
-    "raml-parser": "^0.8.10",
+    "popsicle": "^8.2.0",
+    "popsicle-server": "^2.0.0",
```
Comment (jstoiko) [inline package.json]: move those 2 to `devDependencies`

**Review by jstoiko — CHANGES_REQUESTED** [review]
also add tests for headers

File: package.json (hunk 4 above)
Comment (JanMesaric) [reply package.json]: done

**Review by jstoiko — APPROVED** [review]

File: test/fixtures/example10.raml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  get:
+    description: Returns an organization entity.
+    headers:
+      Authorization:
+        description: |
+           Used to send a valid OAuth 2 access token. Do not use
+           with the "access_token" query string parameter.
+        type: string
+    responses:
+      201:
```
Comment (jstoiko) [inline test/fixtures/example10.raml]: GET vs 201 status code = not the most "RESTful" example. https://tools.ietf.org/html/rfc7231#section-6.3.2

I would change that response status code to 200.

File: test/test10.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+          method: 'GET',
+          url: '/api/organization',
+          headers: {
+            'Authorization': 'Bearer 1234'
+          }
+        }
+      )
+      .use(server(http))
+      .then(function (res) {
+        expect(JSON.parse(res.body)[0].name).to.deep.equal('success')
```
Comment (jstoiko) [inline test/test10.js]: shouldn't this deep equal `{ "success": { "name": "Working" } }`?

File: test/test10.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        expect(res.status).to.equal(201)
```
Comment (jstoiko) [inline test/test10.js]: 200 here

**Review by jstoiko — CHANGES_REQUESTED** [review]

**Review by jstoiko — APPROVED** [review]

File: examples/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+var mockService = require('../bin/osprey-mock-service')
+var express = require('express')
+var parser = require('raml-1-parser')
+var path = require('path')
+var osprey = require('osprey')
+
+var app = express()
+
+parser.loadFile(path.join(__dirname, '/api.raml'))
```
Comment (jstoiko) [inline examples/index.js]: this doesn't work

**Review by jstoiko — CHANGES_REQUESTED** [review]
example needs to be fixed

**Review by jstoiko — APPROVED** [review]

File: package.json [hunk 9]
```diff
@@ -36,19 +36,21 @@
   },
```
Comment (llattes) [inline package.json:36]: I think the version of this service should be bumped, not sure if it is a good idea to maintain as `"0.1.0"` with all these changes.

File: package.json [hunk 10]
```diff
@@ -36,19 +36,21 @@
...
+    "standard": "^8.3.0",
+    "popsicle": "^8.2.0",
+    "popsicle-server": "^2.0.0"
   },
   "dependencies": {
     "finalhandler": "^0.4.0",
     "morgan": "^1.7.0",
     "negotiator": "^0.6.0",
-    "osprey": "^0.3.0",
+    "osprey": "mulesoft/osprey#raml1.0",
```
Comment (llattes) [inline package.json:51]: Once the new version is released, remind to change this reference to the corresponding version (0.4.0 I guess from https://github.com/mulesoft/osprey/pull/116/files)

**Review by llattes — COMMENTED** [review]
Just two comments to consider.

**Comment by W1R3D-Code** [comment]
Can this be merged in and released yet? Even if its as a pre-release? would make it easier to install :octocat:

**Comment by rickydutton** [comment]
+1

**Comment by jstoiko** [comment]
Sorry for the wait. I'd also like this to be merged & released but we're waiting for [an issue ](https://github.com/raml-org/raml-js-parser-2/issues/560) to be fixed in the parser upstream before releasing.

In the meantime, you can install with `npm install git+https://github.com/mulesoft-labs/osprey-mock-service.git#raml1.0`

**Pull request merged**